Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -462,40 +413,53 @@ void CGPathAddPath(CGMutablePathRef path, const CGAffineTransform* transform, CG
 
     // Close the path we're adding and open the path we need to add to.
     FAIL_FAST_IF_FAILED(toAdd->ClosePath());
+
     FAIL_FAST_IF_FAILED(path->PreparePathForEditing());
+    path->EndFigure(D2D1_FIGURE_END_OPEN);
     FAIL_FAST_IF_FAILED(toAdd->GetPathGeometry()->Stream(path->GetGeometrySink().Get()));
+    path->SetCurrentPoint(toAdd->GetCurrentPoint());
+    path->SetStartingPoint(toAdd->GetStartingPoint());
 }
 
 /**
  @Status Interoperable
 */
 void CGPathAddEllipseInRect(CGMutablePathRef path, const CGAffineTransform* transform, CGRect rect) {
     RETURN_IF(!path);
+    CGRect containingRectangle = rect;
+    // First we must transform the coordinates of the rectangle if necessary.
+    if (transform) {
+        containingRectangle = CGRectApplyAffineTransform(rect, *transform);
+    }
```

Comment:
I don't think this is what we want: the transform must apply to the _ellipse_, not to the rect it's contained in.